derive_miniconf/src/lib.rs

```diff
@@ -259,62 +320,34 @@ fn derive_enum(mut typedef: TypeDefinition, data: syn::DataEnum) -> TokenStream
                 *self = miniconf::serde_json_core::from_slice(value)?.0;
                 Ok(())
             }
+
+            fn string_get(&self, mut topic_parts: core::iter::Peekable<core::str::Split<char>>, value: &mut [u8]) -> Result<usize, miniconf::Error> {
+                if topic_parts.peek().is_some() {
+                    // We don't support enums that can contain other values
+                    return Err(miniconf::Error::PathTooLong)
+                }
+
+                miniconf::serde_json_core::to_slice(self, value).map_err(|_| miniconf::Error::SerializationFailed)
+            }
+
+            fn recursive_iter<const TS: usize>(&self, index: &mut [usize], topic: &mut miniconf::heapless::String<TS>) -> Option<()> {
+                if index.len() == 0 {
+                    // I don't expect this to happen...
+                    panic!("index stack too small");
+                    // return None;
+                }
```

Comment:
Given that this is a user error, should we return None instead of panicking? If it's ok to panic here, should be ok to panic when TS is too small.